Develop

AndreyKirshin opened the pull request:
Task2:  
Installed neccessery dependencies
Configured webpack, babel
Add scripts to build and run the app

Commit:
Task1: The project has been set up. Components have been created according the task

Commit:
Tast1: Express has been added to the app

Commit:
Task1: All dependencies have been updated to latest versions
redundant div tag has been raplaced with React.Fragment

Commit:
Task2:
Installed neccessery dependencies
Configured webpack, babel
Add scripts to build and run the app

Commit:
Task2:
Configured test script
Used env variable

SKolevatov commented on .gitignore:
```diff
@@ -59,3 +60,7 @@ typings/
+
+#my conditions
+package-lock.json
+public/index.html
```
you already included public/

AndreyKirshin replied:
done

SKolevatov commented on .gitignore:
```diff
@@ -59,3 +60,7 @@ typings/
 .next
+
+#my conditions
+package-lock.json
```
package-lock is useful file, don't add it in gitignore unless you are sure you want to do it

AndreyKirshin replied:
done

SKolevatov commented on package.json:
```diff
@@ -0,0 +1,40 @@
+  "description": "",
+  "scripts": {
+    "server": "node server/app.js",
+    "build": "webpack --mode production",
```
let's create two scripts for prod and dev builds

AndreyKirshin replied:
done

SKolevatov commented on webpack.config.js:
```diff
@@ -0,0 +1,53 @@
+          loader: 'babel-loader',
+        },
+        {
+          test: /\.jsx$/,
```
instead of adding one more test, you may use
```test: /\.jsx?$/,```
this will cover both `.js` and `.jsx` files

AndreyKirshin replied:
done

Commit:
Task2:
created defferent scripts for prod & dev
added redux, react-redux, react-router
created general test regexp for .js & .jsx files
package-lock.json removed from .gitignore

Commit:
TASK 3:
Created components acoording to markup
Created errorBoundary component
Used smart/dumb components approach

SKolevatov commented on src/client/view/container/container.jsx:
```diff
@@ -0,0 +1,58 @@
+import BuggyButton from '../buggyButton/buggyButton';
+
+export default class Container extends React.Component {
+    constructor() {
```
By the way, note that you can do it a little bit shorter:
```
state = data;
handleChangeSearchValue = (event) => {
        this.setState({searchValue: event.target.value});
    }
....
```
In this case you don't need constructor. That's kind of syntax sugar, but in my opinion looks nicer. Especially when there are a lot of functions to bind.

AndreyKirshin replied:
I used this way

Commit:
TASK3
fixed design of search panel
extracted searchBy buttons to separatrd component
extracted movie into to separated component
